src/ui/public/query_bar/directive/query_popover.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { uiModules } from 'ui/modules';
+import { documentationLinks } from '../../documentation_links/documentation_links';
+import {
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiSwitch,
+  EuiLink,
+  EuiText,
+  EuiSpacer,
+} from '@elastic/eui';
+
+const luceneQuerySyntaxDocs = documentationLinks.query.luceneQuerySyntax;
+const kueryQuerySyntaxDocs = documentationLinks.query.kueryQuerySyntax;
+
+const module = uiModules.get('app/kibana', ['react']);
+module.directive('queryPopover', function (localStorage) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      language: '<',
```

Comment:
I'm not familiar with this, how does it work? And is this why you use `$language` in the query_bar directive's markup?

Reply:
`<` denotes one way binding in angular 1.5+ https://toddmotto.com/one-way-data-binding-in-angular-1-5/#one-way-bindings

`$language` is a completely separate variable, it's part of `onSelectLanguage`'s API as denoted one line 47 https://github.com/elastic/kibana/pull/17232/files/592d9101e15679ee37d57bf30edb2e57ab1f26e4#diff-22f5c29ad70499893c9a6d06ea18320eR47

Reply:
Thanks, totally forgot about one-way binding, and it wasn't particularly easy to google that syntax.

Was there a reason you named the value `$language` here? Does it match some other convention I'm not seeing?

Reply:
It's a convention I picked up from ui-select and some other angular libraries. Variables that get exposed to the expression bound to the `&` scope variable get prefixed with a `$`. I think it's a nice convention because without it it's not obvious in the parent which variables are coming from the parent scope and which are coming from the child's callback. For example, in the [Basic Example](https://github.com/angular-ui/ui-select/wiki/Getting-Started#basic-example) from ui-select here, I can tell that `selected.value` comes from the parent scope but `$select.selected.name` comes from the `ui-select-match` component.